training/Exercise06_Gibbs.ipynb

```diff
@@ -567,23 +639,62 @@
    "source": [
     "# 3 Exploring other priors\n",
     "\n",
-    "Notice that while the sampling went well in the previous example,\n",
-    "the posterior distribution did not match the characteristics of\n",
-    "the exact solution. We can improve this result by switching to a\n",
-    "prior that better matches the exact solution $\\mathbf{x}$.\n",
+    "Suppose the true solution was not modelled well by a GMRF prior. This would for example be the case for a piece-wise constant \"square\" signal. We can quickly try that experiment by generating new data from the test problem with a `square` phantom.\n",
+    "\n",
+    "For illustration we repeat all the code steps from above in a single cell, even though we could have just redefined the posterior.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Deterministic forward model + data\n",
+    "A, y_data, probinfo = Deconvolution1D.get_components(phantom='square')\n",
+    "\n",
+    "# Bayesian model\n",
+    "d = Gamma(1, 1e-4)\n",
+    "s = Gamma(1, 1e-4)\n",
+    "x = GMRF(np.zeros(n), lambda d: d)\n",
+    "y = GaussianCov(A@x, lambda s: 1/s)\n",
     "\n",
-    "One choice is the Laplace difference prior, which assumes a\n",
-    "Laplace distribution for the differences between neighboring\n",
-    "elements of $\\mathbf{x}$. That is,\n",
+    "# Joint distribution\n",
+    "joint = JointDistribution(y, x, d, s)\n",
+    "\n",
+    "# Posterior\n",
+    "posterior = joint(y=y_data)\n",
+    "\n",
+    "# Sampling strategy\n",
+    "sampling_strategy = {\n",
+    "    'x': Linear_RTO,\n",
+    "    'd': Conjugate,\n",
+    "    's': Conjugate\n",
+    "}\n",
+    "\n",
+    "# Gibbs sampler\n",
+    "sampler = Gibbs(posterior, sampling_strategy)\n",
+    "\n",
+    "# Run sampler\n",
+    "samples = sampler.sample(Ns=1000, Nb=200)\n",
+    "\n",
+    "# Plot credible intervals for x\n",
+    "samples['x'].plot_ci(exact=probinfo.exactSolution)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that while the sampling still went well, the posterior distribution did not match the characteristics of the square phantom. We can improve this result by switching to a prior that better matches this kind of signal.\n",
+    "\n",
+    "One choice is the Laplace difference prior, which assumes a Laplace distribution for the differences between neighboring elements of $\\mathbf{x}$. That is we assume\n",
     "\n",
-    "$$\n",
     "\n",
     "\\begin{align*}\n",
     "\\mathbf{x} \\sim \\text{Laplace\\_diff}(\\mathbf{0}, d^{-1}),\n",
```

Comment:
```suggestion
    "\\mathbf{x} \\sim \\text{Laplace\_diff}(\\mathbf{0}, d^{-1}),\n",
```
(I think - it currently shows an additional backslash when viewing in jupyter lab)